Source/Scene/ShadowMap.js

```diff
@@ -307,110 +455,292 @@ define([
-                appearance : new PerInstanceColorAppearance({
-                    translucent : false,
-                    flat : true
-                }),
-                asynchronous : false
-            });
+    var debugCascadeColors = [Color.RED, Color.GREEN, Color.BLUE, Color.MAGENTA];
 
-            //// Create the camera frustum
-            //var nearPlane = frameState.camera.frustum.near;
-            //var farPlane = frameState.camera.frustum.far;
-            //shadowMap._debugCameraFrustum = new Primitive({
-            //    geometryInstances : new GeometryInstance({
-            //        geometry : new BoxOutlineGeometry({
-            //            minimum : new Cartesian3(-0.0, -0.0, -nearPlane),
-            //            maximum : new Cartesian3(0.0, 0.0, -farPlane)
-            //        }),
-            //        attributes : {
-            //            color : ColorGeometryInstanceAttribute.fromColor(Color.RED)
-            //        }
-            //    }),
-            //    appearance : new PerInstanceColorAppearance({
-            //        translucent : false,
-            //        flat : true
-            //    }),
-            //    asynchronous : false
-            //});
+    function applyDebugSettings(shadowMap, frameState) {
+        if (!shadowMap.debugShow) {
+            return;
         }
 
-        //if (!shadowMap.debugFreezeFrame) {
-        //
-        //    //var nearPlane = frameState.camera.frustum.near;
-        //    //var point = Matrix4.multiplyByPoint(frameState.context.uniformState.inverseViewProjection, new Cartesian3(0, 0, -nearPlane), new Cartesian3());
-        //    //console.log(point.x, point.y, point.z);
-        //    //console.log(frameState.camera.positionWC.x, frameState.camera.positionWC.y, frameState.camera.positionWC.z);
-        //    //
-        //    //// TODO : possibly get this from the camera instead
-        //    //var cameraMatrix = frameState.context.uniformState.inverseViewProjection;
-        //    //shadowMap._debugCameraFrustum.modelMatrix = cameraMatrix;
-        //}
-
+        shadowMap._debugLightFrustum = shadowMap._debugLightFrustum || createDebugFrustum(Color.YELLOW);
+        Matrix4.inverse(shadowMap._shadowMapCamera.getViewProjection(), shadowMap._debugLightFrustum.modelMatrix);
         shadowMap._debugLightFrustum.update(frameState);
-        shadowMap._debugLight.update(frameState);
-        //shadowMap._debugCameraFrustum.update(frameState);
+
+        for (var i = 0; i < shadowMap._numberOfCascades; ++i) {
+            shadowMap._debugCascadeFrustums[i] = shadowMap._debugCascadeFrustums[i] || createDebugFrustum(debugCascadeColors[i]);
+            Matrix4.inverse(shadowMap._cascadeCameras[i].getViewProjection(), shadowMap._debugCascadeFrustums[i].modelMatrix);
+            shadowMap._debugCascadeFrustums[i].update(frameState);
+        }
 
         updateDebugShadowViewCommand(shadowMap, frameState);
     }
 
+    function ShadowMapCamera() {
```

Comment:
I actually spent a good amount of time trying to get the shadow map to use `Camera`, but could not find an easy way to give it explicit world space `direction`, `up`, and `right` vectors, without `Camera` recomputing them differently. It seems like the computed light view is not actually affine, which goes against what the `Camera` class wants. For example I had to make this change to get the correct eye position (https://github.com/AnalyticalGraphicsInc/cesium/commit/097c99999830f15180b7140fc177c3df218f5589).